ci/jupyter.js

```diff
@@ -77,8 +107,9 @@ class JupyterServer {
   async writeNotebook(path, notebook) {
     // Once https://github.com/nteract/nteract/pull/3651 is merged, we can use
     // rx-jupyter for writing a notebook to the contents API
+    const apiPath = "/api/contents/";
     const xhr = await ajax({
-      url: `${this.endpoint}/api/contents/${path}`,
+      url: url_path_join(this.endpoint, apiPath, path),
```

Comment:
While developing this is constantly an issue.

Yes, its developer error with consistency around whether routes are defined with `/` at the beginning & end or not. Trying to keep track of it leaves it easy to mess up. 

I'm bringing this pattern over from the jupyter notebook server, where it is used extensively. We're already importing it for the serverside code.

This a huge quality of life improvement, trust me.

Reply:
For reference to the server side code: 

```python
def url_path_join(*pieces):
    """Join components of url into a relative url
    Use to prevent double slash when joining subpath. This will leave the
    initial and final / in place
    """
    initial = pieces[0].startswith('/')
    final = pieces[-1].endswith('/')
    stripped = [s.strip('/') for s in pieces]
    result = '/'.join(s for s in stripped if s)
    if initial: result = '/' + result
    if final: result = result + '/'
    if result == '//': result = '/'
    return result
```

https://github.com/jupyter/notebook/blob/master/notebook/utils.py#L48-L61

Reply:
See PR #88